tests/testsuite/cfg.rs

```diff
@@ -490,3 +490,181 @@ fn cfg_looks_at_rustflags_for_target() {
         execs().with_status(0),
     );
 }
+
+#[test]
+fn target_cfg_bin() {
+    let p = project("target_cfg_bin")
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "bar"
+            version = "0.1.0"
+            authors = []
+
+            [[target.'cfg(target_pointer_width = "1")'.bin]]
+            name = "never"
+            path = "src/never.rs"
+
+            [[target.'cfg(not(target_pointer_width = "1"))'.bin]]
+            name = "always"
+            path = "src/always.rs"
+        "#,
+        )
+        .file("src/main.rs", &main_file("\"main\"", &[]))
+        .file("src/never.rs", &main_file("\"never\"", &[]))
+        .file("src/always.rs", &main_file("\"always\"", &[]))
+        .build();
+
+    assert_that(
+        p.cargo("build").arg("-v"),
+        execs().with_status(0)
+            // TODO FITZGEN: I don't know why this assertion fails...
```

Comment:
You'll want to write down `autobins = true` in the manifest above to get this working.